src/test/rgw/test_rgw_reshard_wait.cc

```diff
@@ -67,10 +67,10 @@ TEST(ReshardWait, wait_yield)
       EXPECT_EQ(0, waiter.wait(optional_yield{context, yield}));
     });
 
+  const ceph::real_time start = ceph::real_clock::now();
```

Comment:
i am confused. i feel that the original version is more correct. `RGWReshardWait` won't be executed unless `io_context::run_one()` is called. so elapsed time should be greater or equal to how long `waiter.wait()` waits.

probably a mono clock is better?

Reply:
`boost::asio::spawn()` schedules our lambda as a handler on the `context`, and the initial call to `context.poll()` is what executes that handler - so `RGWReshardWait::wait()` starts the timer here, and suspends the coroutine inside of `waiter.timer.async_wait(yield[ec]);`

when that timer expires, a new handler is scheduled on `context` that will resume the coroutine in `async_wait()` and allow the `RGWReshardWait::wait()` call to complete. it's that second handler that the call to `context.run_one()` is blocking on

since `context.poll()` is where `RGWReshardWait`'s timer is started, the test needs to start its timer before that